code/modules/surgery/organs/internal/cyberimp/augments_chest.dm

```diff
@@ -52,47 +52,86 @@
 	slot = ORGAN_SLOT_HEART_AID
 	var/revive_cost = 0
 	var/reviving = FALSE
+	/// revival/defibrillation possibility flag that gathered from owner's .can_defib() proc
+	var/can_defib_owner
 	COOLDOWN_DECLARE(reviver_cooldown)
 
+/obj/item/organ/internal/cyberimp/chest/reviver/on_death(seconds_per_tick, times_fired)
+	try_heal() // Allowes implant to work even on dead people
 
 /obj/item/organ/internal/cyberimp/chest/reviver/on_life(seconds_per_tick, times_fired)
+	try_heal()
+
+/obj/item/organ/internal/cyberimp/chest/reviver/proc/try_heal()
 	if(reviving)
 		switch(owner.stat)
-			if(UNCONSCIOUS, HARD_CRIT, SOFT_CRIT)
-				addtimer(CALLBACK(src, PROC_REF(heal)), 3 SECONDS)
-			else
+			if(CONSCIOUS)
```

Comment:
Get rid of this switch, just do `if(stat == CONSCIOUS) else`

Reply:
Done